pkg/controllers/dynakube/activegate/conditions/conditions.go

```diff
@@ -0,0 +1,70 @@
+package conditions
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ReasonCreated         string = "Created"
+	ReasonError           string = "Error"
+	ReasonUnexpectedError string = "UnexpectedError"
+	ReasonUpToDate        string = "UpToDate"
+)
+
+// ActiveGate related conditions.
+const (
+	ActiveGateConnectionInfoConditionType string = "ActiveGateConnectionInfo"
+	ActiveGateStatefulSetConditionType    string = "ActiveGateStatefulSet"
+	ActiveGateVersionConditionType        string = "ActiveGateVersion"
+)
+
+func SetActiveGateConnectionInfoCondition(conditions *[]metav1.Condition, err error) {
+	if err != nil {
+		meta.SetStatusCondition(conditions, metav1.Condition{
+			Type:    ActiveGateConnectionInfoConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  ReasonUnexpectedError,
+			Message: err.Error(),
+		})
+
+		return
+	}
+
+	meta.SetStatusCondition(conditions, metav1.Condition{
+		Type:   ActiveGateConnectionInfoConditionType,
+		Status: metav1.ConditionTrue,
+		Reason: ReasonCreated,
+	})
+}
+
+func SetActiveGateStatefulSetErrorCondition(conditions *[]metav1.Condition, err error) {
+	if err != nil {
+		meta.SetStatusCondition(conditions, metav1.Condition{
+			Type:    ActiveGateStatefulSetConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  ReasonError,
+			Message: err.Error(),
+		})
+	}
+}
+
+func SetActiveGateVersionCondition(conditions *[]metav1.Condition, version string, err error) {
+	if err != nil {
+		meta.SetStatusCondition(conditions, metav1.Condition{
+			Type:    ActiveGateVersionConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  ReasonError,
+			Message: err.Error(),
+		})
+
+		return
+	}
+
+	meta.SetStatusCondition(conditions, metav1.Condition{
+		Type:    ActiveGateVersionConditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  ReasonUpToDate,
+		Message: version,
+	})
+}
```

Comment:
could be in the version package, and it could be private ([we already have similar ones for OneAgent, so try to reuse what you can](https://github.com/Dynatrace/dynatrace-operator/blob/37afc6130005c92ba0ff4fddb503292be48269aa/pkg/controllers/dynakube/version/conditions.go#L27-L27))